subsys/net/lib/tls_credentials/tls_credentials_digest_raw.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Internal helper function for generating digests for raw credentials.
+ */
+
+#ifndef __TLS_DIGEST_RAW_H
+#define __TLS_DIGEST_RAW_H
+
+#include <zephyr/net/tls_credentials.h>
+#include "tls_internal.h"
+
+/* Common version of credential_digest that raw credentials backends can use. */
+int credential_digest_raw(struct tls_credential *credential, void *dest, size_t *len);
```

Comment:
Good catch, I think that was an oversight on my part. I vacillated between defining in `tls_internal` and in its own header file. I landed on using a separate header file because the underlying C file may not always be included in the build, depending on any future prospective back-ends we add.